test/unit/multiplication/test_multiplication_general.cpp

```diff
@@ -200,6 +200,7 @@ TYPED_TEST(GeneralSubMultiplicationDistTestMC, CorrectnessDistributed) {
       const TypeParam beta = TypeUtilities<TypeParam>::element(-2.6, .7);
       testGeneralSubMultiplication<TypeParam, Backend::MC, Device::CPU>(comm_grid, a, b, alpha, beta, m,
                                                                         mb);
+      pika::threads::get_thread_manager().wait();
```

Comment:
Perhaps you added these (this and the next one) because of the missing `.clone()` (see my comment above in the algorithm overloaded call)?

Reply:
I removed here :  https://github.com/eth-cscs/DLA-Future/pull/719/commits/0189f92a49066f3374d6a1c0f6580549e87e50f5